src/repo.rs

```diff
@@ -356,6 +419,32 @@ impl<T: NetworkAdapter + 'static> Repo<T> {
                     panic!("Document closed with outstanding handles.");
                 }
             }
+            RepoEvent::ConnectNetworkAdapter(repo_id, adapter) => {
+                self.network_adapters
+                    .entry(repo_id.clone())
+                    .and_modify(|_| {
+                        // TODO: close the existing stream/sink?
+                    })
+                    .or_insert(adapter);
+
+                // Try to sync all docs we know about.
```

Comment:
This is basically telling the newly connected repo all the documents we have locally, by starting the sync. Obviously we could send Id's instead or something, but for now keeping it simple.